src/lib.rs

```diff
@@ -4,11 +4,18 @@ use ispc::WeightCollection;
 
 mod ispc;
 
-pub trait ImagePixelFormat {
-    fn num_channel_in_memory(&self) -> usize;
-    fn channel_size_in_bytes(&self) -> usize;
-
-    fn pixel_size(&self) -> usize {
+pub trait ImagePixelFormat: Copy {
+    /// Returns the number of channels that an image of this format would have in memory.
+    /// For example, while a normal map of format [`NormalMapFormat::R8g8TangentSpaceReconstructedZ`] would still have 3 channels when sampled,
```

Comment:
"When sampled" here refers to when `ispc-downsampler` reads it?

Reply:
Depends on your definition of sample. The downsampler will read only two channels, because the format has only two channels in memory, and then reconstruct the last one, making it for 3 channels that are used in the downsampling process.

Reply:
So that is `ispc-downsampler`.